zebra/zebra_pw.c

```diff
@@ -153,6 +153,7 @@ void zebra_pw_update(struct zebra_pw *pw)
 {
 	if (zebra_pw_check_reachability(pw) < 0) {
 		zebra_pw_uninstall(pw);
+		zebra_pw_install_failure(pw);
```

Comment:
> The original code shows the intention of retrying,
> but the code was missing

@pacovn I believe you're referring to this comment:
```
/* wait for NHT and try again later */
```

Essentially, if `zebra_pw_check_reachability()` fails, it means that the pseudowire remote endpoint isn't reachable (i.e. there's no LSP to it). In this case, we need to wait for other NHT events until the reachability is satisfied.

If we call `zebra_pw_install_failure()` when the pseudowire remote endpoint isn't reachable, we'll try to install it 30 seconds later regardless if the reachability condition was satisfied or not. This can lead to unusable pseudowires being installed in the forwarding plane.

If the reconvergence isn't working as expected, I think we need to look for a bug in the NHT code instead. I've tried to reproduce this problem using my netgen VPLS topologies but everything seems to be working normally. The pseudowires are always uninstalled/reinstalled after killing and restarting isisd somewhere in the network. But you might be doing something different though.

Using `zebra_pw_install_failure()` is wrong in this case since this function was designed to retry installing a valid pseudowire in the forwarding plane. A pseudowire installation can fail, for example, when the hardware runs out of resources. Then retrying the installation every 30 seconds is a sensible thing to do. This function shouldn't be used in other circumstances.